Userland/Libraries/LibCore/Stream.cpp

```diff
@@ -691,6 +691,34 @@ ErrorOr<pid_t> LocalSocket::peer_pid() const
 #endif
 }
 
+ErrorOr<uid_t> LocalSocket::peer_uid() const
+{
+#ifdef AK_OS_MACOS
+    uid_t uid;
+    socklen_t uid_size = sizeof(uid);
+#elif defined(AK_OS_FREEBSD)
+    struct xucred creds = {};
+    socklen_t creds_size = sizeof(creds);
+#elif defined(AK_OS_OPENBSD)
+    struct sockpeercred creds = {};
+    socklen_t creds_size = sizeof(creds);
+#else
+    struct ucred creds = {};
+    socklen_t creds_size = sizeof(creds);
+#endif
+
```

Comment:
From a quick (and half-sober) glance; couldn't this be simplified as?
```cpp
#ifdef AK_OS_MACOS
    uid_t uid;
    socklen_t uid_size = sizeof(uid);
#else
#    ifdef AK_OS_FREEBSD
    struct xucred creds = {};
#    elif defined(AK_OS_OPENBSD)
    struct sockpeercred creds = {};
#    else
    struct ucred creds = {};
#    endif
    socklen_t creds_size = sizeof(creds);
#    endif
#endif
```

as to not duplicate `socklen_t creds_size = sizeof(creds);`. Though the current solution might be more redable.